apps/run.go

```diff
@@ -322,42 +322,43 @@ func (runCtx *runContext) connectToRunServer(ctx context.Context) (*http.Respons
 		return nil, nil, errgo.New("No attach URL to connect to")
 	}
 
-	req, err := http.NewRequest("CONNECT", runCtx.attachURL, nil)
+	url, err := url.Parse(runCtx.attachURL)
 	if err != nil {
 		return nil, nil, errgo.Mask(err, errgo.Any)
 	}
-	token, err := runCtx.scalingoClient.GetAccessToken(ctx)
 
+	// Establish an initial TCP connection
+	conn, err := net.Dial("tcp", url.Host)
 	if err != nil {
-		return nil, nil, errgo.Notef(err, "fail to generate auth")
+		return nil, nil, errgo.Mask(err, errgo.Any)
 	}
-	req.SetBasicAuth("", token)
 
-	url, err := url.Parse(runCtx.attachURL)
+	// Wrap with TLS if using HTTPS
+	if url.Scheme == "https" {
+		tlsConfig := config.TLSConfig.Clone()
+		tlsConfig.ServerName = strings.Split(url.Host, ":")[0]
+		conn = tls.Client(conn, tlsConfig)
+	}
+
+	req, err := http.NewRequest("CONNECT", runCtx.attachURL, nil)
 	if err != nil {
 		return nil, nil, errgo.Mask(err, errgo.Any)
 	}
 
-	dial, err := net.Dial("tcp", url.Host)
+	token, err := runCtx.scalingoClient.GetAccessToken(ctx)
 	if err != nil {
-		return nil, nil, errgo.Mask(err, errgo.Any)
+		return nil, nil, errgo.Notef(err, "fail to generate auth")
```

Comment:
issue: another thing which changed about our error message is that we no longer want to include words such as "fail to" in the wrapping error message
```suggestion
		return nil, nil, errors.Wrap(ctx, err, "get access token")
```